apps/site/lib/site_web/controllers/trip_compare_controller.ex

```diff
@@ -0,0 +1,565 @@
+      select: "plan-time-select",
+      label: "plan-time-label"
+    },
+    title: "trip-plan-departure-title"
+  }
+
+  def index(conn, %{"plan" => %{"to" => _to, "from" => _fr} = plan}) do
+    conn
+    |> assign(:expanded, conn.query_params["expanded"])
+    |> render_plan(plan)
+  end
+
+  def index(conn, _params) do
+    render(conn, :index)
+  end
+
+  def to(conn, %{"plan" => _plan} = params) do
+    redirect(conn, to: trip_compare_path(conn, :index, Map.delete(params, "address")))
+  end
+
+  def to(conn, %{
+        "address" => address
+      }) do
+    if String.match?(address, ~r/^(\-?\d+(\.\d+)?),(\-?\d+(\.\d+)?),.*$/) do
+      [latitude, longitude, name] = String.split(address, ",", parts: 3)
+      # Avoid extra geocode call, just use these coordinates
+      destination = %TripPlan.NamedPosition{
+        latitude: String.to_float(latitude),
+        longitude: String.to_float(longitude),
+        name: name,
+        stop_id: nil
+      }
+
+      do_to(conn, destination)
+    else
+      updated_address = Geocode.check_address(address, @options)
+
+      case TripPlan.geocode(updated_address) do
+        {:ok, geocoded_to} ->
+          do_to(conn, geocoded_to)
+
+        {:error, _} ->
+          # redirect to the initial index page
+          redirect(conn, to: trip_compare_path(conn, :index))
+      end
+    end
+  end
+
+  defp do_to(conn, destination) do
+    # build a default query with a pre-filled 'to' field:
+    query = %Query{
+      to: destination,
+      time: {:error, :unknown},
+      from: {:error, :unknown}
+    }
+
+    now = Util.now()
+
+    # build map information for a single leg with the 'to' field:
+    map_data =
+      TripPlanMap.itinerary_map([
+        %Leg{
+          from: nil,
+          to: destination,
+          mode: %PersonalDetail{},
+          description: "",
+          start: now,
+          stop: now,
+          name: "",
+          long_name: "",
+          type: "",
+          url: "",
+          polyline: ""
+        }
+      ])
+
+    %{markers: [marker]} = map_data
+    to_marker = %{marker | id: "B"}
+    map_info_for_to_destination = %{map_data | markers: [to_marker]}
+
+    conn
+    |> assign(:query, query)
+    |> assign(:map_data, map_info_for_to_destination)
+    |> render(:index)
+  end
+
+  defp get_route(link) do
+    if is_bitstring(link.text) do
+      link.text
+    else
+      link.text |> List.to_string()
+    end
+  end
+
+  defp filter_duplicate_links(related_links) do
+    Enum.map(related_links, fn x -> Enum.uniq_by(x, fn y -> get_route(y) end) end)
+  end
+
+  @spec render_plan(Plug.Conn.t(), map) :: Plug.Conn.t()
+  defp render_plan(conn, plan) do
+    [current_results, new_results] =
+      Util.async_with_timeout(
+        [
+          fn -> get_itineraries(conn, plan, :root_url, "CURRENT") end,
+          fn -> get_itineraries(conn, plan, :root_url_compare, "NEW") end
+        ],
+        {:error, :timeout},
+        __MODULE__
+      )
+
```

Comment:
blocking: let's add a check for whether we get that `{:error, :timeout}` and handle it properly.

one way you can easily trigger this locally is by using `OPEN_TRIP_PLANNER_URL=http://otp-dev-green.mbtace.com` (which we haven't yet fixed)

Reply:
what would be the right way to handle it? I think existing trip plan controller is not checking for it either.. https://github.com/mbta/dotcom/blob/master/apps/site/lib/site/trip_plan/query.ex#L76-L84

Reply:
I'm not sure exactly where/how it happens, but when the current trip planner times out it just takes you back to the trip planner page. I wouldn't be opposed to having the same happen in this case.

Reply:
👍 ok, I've added a check and I believe it now behaves in the same way